tool_path_planner/src/halfedge_boundary_finder.cpp

```diff
@@ -207,98 +207,16 @@ bool applyEqualDistance_original(const pcl::PointCloud<pcl::PointNormal>& in, pc
     v_1 = p_mid.getVector3fMap() - p_start.getVector3fMap();
   }
 
-  if(out.size() < 2)
-  {
-    CONSOLE_BRIDGE_logError("Failed to add points");
-    return false;
-  }
-
   // add last point if not already there
   if((out.back().getVector3fMap() - in.back().getNormalVector3fMap()).norm() > MIN_POINT_DIST_ALLOWED)
   {
     out.push_back(in.back());
   }
 
-  // recovering normals now
-  std::vector<int> k_indices;
-  std::vector<float> k_sqr_distances;
-  PointXYZ query_point;
-  for(auto& p : out)
-  {
-    k_indices.resize(1);
-    k_sqr_distances.resize(1);
-    std::tie(query_point.x, query_point.y, query_point.z) = std::make_tuple(p.x, p.y, p.z);
-    if(kdtree.nearestKSearch(query_point,1,k_indices, k_sqr_distances) < 1)
-    {
-      CONSOLE_BRIDGE_logError("No nearest points found");
-      return false;
-    }
-
-    const auto& pnormal = in[k_indices.front()];
-    std::tie(p.normal_x, p.normal_y, p.normal_z) = std::make_tuple(
-        pnormal.normal_x, pnormal.normal_y, pnormal.normal_z);
-  }
-  return true;
-}
-
-/**
- * @brief forces the required point spacing by computing new points along the curve
- * @param in    The input cloud
- * @param out   The output cloud
- * @param dist  The required point spacing distance
- * @return True on success, False otherwise
- */
-bool applyEqualDistance(const pcl::PointCloud<pcl::PointNormal>& in, pcl::PointCloud<pcl::PointNormal>& out,double dist)
```

Comment:
change the name of this function to applyEqualPerimeterDistance, and keep it around even though nobody is calling it at this point. There may be some reasons to use it. For example, if the perimeter is more bumpy, such an algorithm would tend to slow down. Also, if one had a compliant tool riding along the surface, one would want to move at a constant edge distance, not a constant Cartesian velocity.

Reply:
There's already a function that does this and it is selectable using the method field in the configuration so no need to keep this function around.